flink-runtime/src/main/java/org/apache/flink/runtime/state/DefaultOperatorStateBackend.java

```diff
@@ -497,6 +497,15 @@ public String toString() {
 
 			return partitionOffsets;
 		}
+
+		@Override
+		public void update(List<S> values) throws Exception {
+			internalList.clear();
+
+			if (values != null && !values.isEmpty()) {
```

Comment:
nit: I think the second check is not required and makes the statement more verbose for no good reason.

Reply:
it makes a difference as we chatted about empty list case